File: .circle_ci/fake-sendmail.rb
#!/usr/bin/ruby
path = "/tmp"
Dir.mkdir(path) if !File.exists?(path)
File.open("#{path}/mailpoet-#{Time.now.to_f}.txt", "w") do |f|
    sleep 5
    f.puts ARGV.inspect
    $stdin.each_line { |line| f.puts line }
end

File: .circleci/fake-sendmail.php
#!/usr/local/bin/php
<?php
$path = "/tmp/fake-mailer";
if (!file_exists($path)) {
  mkdir($path);
}
$filename = $path . '/mailpoet-' . microtime(true) . '.txt';
$fileHandle = fopen($filename, "w");

// phpcs:ignore Squiz.PHP.DiscouragedFunctions
$callArguments = print_r($argv, true) . "\n";
fwrite($fileHandle, $callArguments);

while ($line = fgets(STDIN)) {
  fwrite($fileHandle, $line);
}

fclose($fileHandle);
